contractcourt/htlc_timeout_resolver.go

```diff
@@ -866,9 +937,12 @@ func (h *htlcTimeoutResolver) IsResolved() bool {
 
 // report returns a report on the resolution state of the contract.
 func (h *htlcTimeoutResolver) report() *ContractReport {
-	// If the sign details are nil, the report will be created by handled
-	// by the nursery.
-	if h.htlcResolution.SignDetails == nil {
+	// If the sign details are nil, the report will be created by handled by
+	// the nursery.Sign details are only non-nil for anchor channels. We
+	// then further restrict this path by ensuring we'll only return nil for
+	// non-anchor local commitmetn sweeps.
+	if h.htlcResolution.SignDetails == nil && h.
```

Comment:
Trying to digest the change - I think it means we won't create reports here for spending of outputs from a local commitment with a legacy channel type, but for direct spending, aka, spending of the remote commitment, we'd always handle it here no matter it's anchor or legacy?

Reply:
Yeah I think we can make this section a bit more readable if we break out the various `!=` checks into a new enum that explains the type of resolution we're doing. 

I think would look something like: 
```
var timeoutResolution uint8 

const (
    anchorLocalResolution = iota // h.htlcResolution.SignDetails != nil
    remoteResolution  // h.htlcResolution.SignedTimeoutTx == nil
    legacyNurseryResolution // h.htlcResolution.SignDetails == nil
)
```

Reply:
> I think it means we won't create reports here for spending of outputs from a local commitment with a legacy channel type, but for direct spending, aka, spending of the remote commitment, we'd always handle it here no matter it's anchor or legacy?

With this change, the goal is that only non-anchor, local force close sweeps go through the nursery. 

The CNCT now handles generating the report when the remote party has gone to chain and we want to sweep their output.

The nursery handles the report for legacy channels, where we've gone to chain ourselves.

Reply:
An enum would help a lot with readability and reducing confusion in the future.